connaisseur/workload_object.py

```diff
@@ -70,38 +71,48 @@ def parent_images(self):
                     message=msg, parent_kind=kind, parent_name=name, parent_uid=uid
                 )
 
-            parent_images += WorkloadObject(parent, self.namespace).container_images
-        return parent_images
+            parent_containers.update(WorkloadObject(parent, self.namespace).containers)
+        return parent_containers
 
     @property
-    def container_images(self):
+    def containers(self):
         spec = self._spec["template"]["spec"]
-        return [
-            container["image"]
-            for container in (spec["containers"] + spec.get("initContainers", []))
-        ]
+        return {
+            (container_type, index): Image(container["image"])
+            for container_type in ["containers", "initContainers"]
+            for index, container in enumerate(spec.get(container_type, []))
+        }
```

Comment:
This code is often duplicated in child classes, with the only difference being the value of `spec`.

I would change line 83 so that it uses `self.spec`, and then remove all of the `spec =` assignments (including in child classes), and instead make a property called `spec` which returns the right-hand side value that was previously assigned. This removes the code duplication, but you still have the flexibility and on-demand evaluation.

Reply:
did as you asked